mscore/resourceManager.cpp

```diff
@@ -373,6 +970,232 @@ void ResourceManager::downloadExtension()
+      }
+
+QString ResourceManager::downloadPluginPackage(PluginPackageDescription& desc, const QString& page_url)
+      {
+      DownloadUtils package(QNetworkRequest::RedirectPolicy::NoLessSafeRedirectPolicy, this);
+      package.setTarget(desc.direct_link);
+      // TODO: try to get extension name from direct_link, and add it to localPath
+      QString localPath = dataPath + "/plugins/" + filenameBaseFromPageURL(page_url);
+      if (desc.source == GITHUB || desc.source == GITHUB_RELEASE)
+            localPath += ".zip";
+      else if (desc.source == ATTACHMENT)
+            localPath += "."+QFileInfo(desc.direct_link).suffix();
+      QDir().mkpath(dataPath + "/plugins");
+      package.setLocalFile(localPath);
+      package.download();
+      package.saveFile();
+      // update the timestamp again
+      desc.last_modified = package.getHeader(QNetworkRequest::LastModifiedHeader).toDateTime();
+      return localPath;
+      }
+
+//---------------------------------------------------------
+//   downloadInstallPlugin
+//---------------------------------------------------------
+
+void ResourceManager::downloadInstallPlugin()
+      {
+      QPushButton* button = static_cast<QPushButton*>(sender());
+      button->setEnabled(false);
+      button->setText(tr("Analyzing..."));
+      QString page_url = button->property("page_url").toString();
+      PluginPackageDescription new_package;
+      new_package.package_name = button->property("name").toString();
+      analyzePluginPage("https://musescore.org" + page_url, new_package);
+      if (new_package.source == UNKNOWN) {
+            button->setText("Unknown plugin source.");
+            button->setEnabled(true);
+            return;
+            }
+      button->setText(tr("Downloading..."));
+      QString localPath = downloadPluginPackage(new_package, page_url);
+
+      if (installPluginPackage(localPath, new_package)) {
+            // add the new description item to the map
+            pluginDescriptionMap[page_url] = new_package;
+            refreshPluginButton(button->property("row").toInt());
+            writePluginPackages();
+            displayPlugins();
+            }
+      else {
+            button->setText("Bad archive. Try again");
+            button->setEnabled(true);
+            return;
+            }
+      }
+
+void ResourceManager::uninstallPluginPackage() 
+      {
+      QPushButton* button = static_cast<QPushButton*>(sender());
+      const QString& url = button->property("page_url").toString();
+      if (!pluginDescriptionMap.contains(url))
+            return;
+      PluginPackageDescription& desc = pluginDescriptionMap[url];
+      // remove qml files first(which may be outside the folder of this plugin)
+      for (QString& path : desc.qml_paths)
+            QFile::remove(path);
+      // then remove the folder
+      QDir d(desc.dir);
+      if (!d.removeRecursively()) {
+            button->setText("Failed, try again");
```

Comment:
here too

Reply:
Thanks. Fixed in https://github.com/musescore/MuseScore/pull/4833/commits/1d26ad066f4fdc3c5897d2e6c467a6de02ce32f4.